lib/supabase/account_organization_ids/addAccountToOrganization.ts

```diff
@@ -0,0 +1,52 @@
+import supabase from "@/lib/supabase/serverClient";
+
+/**
+ * Add an account to an organization
+ * Used for auto-assignment on login based on email domain
+ *
+ * @param accountId - The account ID
+ * @param organizationId - The organization ID to add them to
+ * @returns The created record ID, or null if failed/already exists
+ */
+export async function addAccountToOrganization(
+  accountId: string,
+  organizationId: string
+): Promise<string | null> {
+  if (!accountId || !organizationId) return null;
+
+  try {
+    // Check if already a member
+    const { data: existing } = await supabase
+      .from("account_organization_ids")
+      .select("id")
+      .eq("account_id", accountId)
+      .eq("organization_id", organizationId)
+      .single();
+
+    if (existing) {
+      // Already a member, return existing ID
+      return existing.id;
+    }
```

Comment:
KISS principle
- actual: adding multiple supabase queries in a single function.
- required: remove the first `select`. Instead, let the `insert` call fail for existing associations.

Reply:
If the goal of this function is also to act as a `select` it does not follow the Single Responsibility Principle.